tests/Aspire.Oracle.EntityFrameworkCore.Tests/ConformanceTests_Pooling.cs

```diff
@@ -0,0 +1,122 @@
+        "Microsoft.EntityFrameworkCore.Query",
+        "Microsoft.EntityFrameworkCore.Database.Transaction",
+        "Microsoft.EntityFrameworkCore.Database.Connection",
+        "Microsoft.EntityFrameworkCore.Model",
+        "Microsoft.EntityFrameworkCore.Model.Validation",
+        "Microsoft.EntityFrameworkCore.Update",
+        "Microsoft.EntityFrameworkCore.Migrations"
+    };
+
+    protected override string JsonSchemaPath => "src/Components/Aspire.Oracle.EntityFrameworkCore/ConfigurationSchema.json";
+
+    protected override string ValidJsonConfig => """
+        {
+          "Aspire": {
+            "Oracle": {
+              "EntityFrameworkCore": {
+                "ConnectionString": "YOUR_CONNECTION_STRING",
+                "HealthChecks": false,
+                "DbContextPooling": true,
+                "Tracing": true,
+                "Metrics": true
+              }
+            }
+          }
+        }
+        """;
+
+    protected override (string json, string error)[] InvalidJsonToErrorMessage => new[]
+        {
+            ("""{"Aspire": { "Oracle": { "EntityFrameworkCore":{ "MaxRetryCount": "5"}}}}""", "Value is \"string\" but should be \"integer\""),
+            ("""{"Aspire": { "Oracle": { "EntityFrameworkCore":{ "HealthChecks": "false"}}}}""", "Value is \"string\" but should be \"boolean\""),
+            ("""{"Aspire": { "Oracle": { "EntityFrameworkCore":{ "ConnectionString": "", "DbContextPooling": "Yes"}}}}""", "Value is \"string\" but should be \"boolean\"")
+        };
+
+    protected override void PopulateConfiguration(ConfigurationManager configuration, string? key = null)
+        => configuration.AddInMemoryCollection(new KeyValuePair<string, string?>[1]
+        {
+            new KeyValuePair<string, string?>("Aspire:Oracle:EntityFrameworkCore:ConnectionString", ConnectionString)
+        });
+
+    protected override void RegisterComponent(HostApplicationBuilder builder, Action<OracleEntityFrameworkCoreSettings>? configure = null, string? key = null)
+        => builder.AddOracleDatabaseDbContext<TestDbContext>("orclconnection", configure);
+
+    protected override void SetHealthCheck(OracleEntityFrameworkCoreSettings options, bool enabled)
+        => options.HealthChecks = enabled;
+
+    protected override void SetTracing(OracleEntityFrameworkCoreSettings options, bool enabled)
+        => options.Tracing = enabled;
+
+    protected override void SetMetrics(OracleEntityFrameworkCoreSettings options, bool enabled)
+        => options.Metrics = enabled;
+
+    protected override void TriggerActivity(TestDbContext service)
+    {
+        if (service.Database.CanConnect())
+        {
+            service.Database.EnsureCreated();
+        }
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "EF1001:Internal EF Core API usage.", Justification = "Required to verify pooling without touching DB")]
```

Comment:
Can you move this to a `#pragma` only around the code that needs it?

Reply:
Done